router/functionHandler.go

```diff
@@ -154,23 +156,37 @@ func (roundTripper RetryingRoundTripper) RoundTrip(req *http.Request) (resp *htt
 			log.Printf("Calling getServiceForFunction for function: %s", roundTripper.funcHandler.function.Name)
 
 			// send a request to executor to specialize a new pod
-			service, err := roundTripper.funcHandler.executor.GetServiceForFunction(
+			resp, err := roundTripper.funcHandler.executor.GetServiceForFunction(
 				roundTripper.funcHandler.function)
+
 			if err != nil {
-				log.Printf("Err from GetServiceForFunction : %v", err)
+				log.Debugf("Err from GetServiceForFunction : %v", err)
 				// We might want a specific error code or header for fission failures as opposed to
 				// user function bugs.
 				return nil, err
 			}
 
+			if resp.StatusCode != http.StatusOK {
+				if roundTripper.funcHandler.envType != "production" && len(req.Header.Get(fission.FissionDebugHeader)) != 0 {
+					return resp, nil
+				}
+
+				return nil, fmt.Errorf("Internal Error")
+			}
+
+			svcName, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
```

Comment:
The errors of the `io` package are not very contextual. I would be in favour of just wrapping the `err` with an error explaining that it failed to read the body of the GetService request?